File: packages/axon-core-ts/src/persona.ts
/** Persona model used by FieldOps for ESS commissioning workflows. */

export type InteractionLevel = "intentional" | "incidental" | "none";

export interface Persona {
  id: string;
  name: string;
  role: string;
  interactionLevel: InteractionLevel;
  goals: string[];
  pains: string[];
  tools: string[];
  motivationalDrivers?: string[];
  knowledge?: string[];
  workhours?: string;
}

File: packages/axon-core-py/src/axon_core/persona.py
from __future__ import annotations
from typing import Literal, Optional
from pydantic import BaseModel, Field

InteractionLevel = Literal["intentional", "incidental", "none"]


class Persona(BaseModel):
    id: str
    name: str
    role: str
    interactionLevel: InteractionLevel
    goals: list[str] = Field(default_factory=list)
    pains: list[str] = Field(default_factory=list)
    tools: list[str] = Field(default_factory=list)
    motivationalDrivers: Optional[list[str]] = None
    knowledge: Optional[list[str]] = None
    workhours: Optional[str] = None
